src/CodeGen/ODataT4CodeGenerator.cs

```diff
@@ -3850,7 +3865,23 @@ internal override void WriteGeneratedEdmModel(string escapedEdmxString)
                 global::System.Xml.XmlReader reader = CreateXmlReader(Edmx);
                 try
                 {
-                    return global::Microsoft.OData.Edm.Csdl.CsdlReader.Parse(reader);
+                    global::Microsoft.OData.Edm.Csdl.CsdlReaderSettings edmxReaderSettings = new global::Microsoft.OData.Edm.Csdl.CsdlReaderSettings()
+                    {
+                        IgnoreUnexpectedAttributesAndElements = ");
+this.Write(this.context.IgnoreUnexpectedElementsAndAttributes.ToString().ToLower());
+this.Write(@"
+                    };
+                    
+                    global::System.Collections.Generic.IEnumerable<global::Microsoft.OData.Edm.Validation.EdmError> errors;
+                    global::System.Collections.Generic.IEnumerable<global::Microsoft.OData.Edm.IEdmModel> references = global::System.Linq.Enumerable.Empty<global::Microsoft.OData.Edm.IEdmModel>();
+                    global::Microsoft.OData.Edm.IEdmModel edmModel;
+                    
+                    if (!global::Microsoft.OData.Edm.Csdl.CsdlReader.TryParse(reader, references, edmxReaderSettings, out edmModel, out errors))
+                    {
+                        throw new global::System.InvalidOperationException(""Error while parsing CSDL."");

```

Comment:
>global::System.InvalidOperationException(""Error while parsing CSDL."") [](start = 34, length = 71)

"Error while parsing CSDL." looks meaningless.
Maybe better to include the errors into the exception?

Reply:
Thanks, changed to:

```
if (!global::Microsoft.OData.Edm.Csdl.CsdlReader.TryParse(reader, references, edmxReaderSettings, out edmModel, out errors))
{
    global::System.Collections.Generic.List<Microsoft.OData.Edm.Validation.EdmError> errorMessages = new global::System.Collections.Generic.List<Microsoft.OData.Edm.Validation.EdmError>(errors);
    throw new global::System.InvalidOperationException(string.Join(";", errorMessages.ConvertAll(e => e.ErrorMessage).ToArray()));
}
```